.github/workflows/e2e-matrix-builder.yml

```diff
@@ -10,88 +10,63 @@ on:
-            const chunk = (arr, size) => {
-              const chunks = [];
-              const n = Math.ceil(arr.length / size);
-              for (let i = 0; i < n; i++) {
-                chunks.push(arr.slice(i * size, (i + 1) * size));
-              }
-              return chunks;
-            };
-
             const java = 21;
             const defaultRunner = "ubuntu-22.04";
 
-            // these are special and shouldn't be chunked out arbitrarily
-            const blacklist = [
-              "/embedding-sdk/",
-              "/cross-version/"
-            ];
-
             const defaultOptions = {
               "java-version": java,
               runner: defaultRunner,
-              specs: "./e2e/test/scenarios/**/*.cy.spec.*",
               edition: "ee",
             };
 
+            const allSpecs = "./e2e/test/scenarios/**/*.cy.spec.*";
+
+            // see config.js function getSplittableSpecs for logic that excludes some tests from auto splitting
             const specialTests = [
               { name: "embedding-sdk", specs: "./e2e/test/scenarios/embedding-sdk/**.cy.spec.*" },
-              { name: "oss-subset", edition: 'oss', tags: "@OSS" },
-              { name: "slow", tags: "@slow+-@flaky" },
-              { name: "mongo", tags: "@mongo" },
+              { name: "oss-subset", edition: 'oss', tags: "@OSS", specs: allSpecs },
+              { name: "mongo", tags: "@mongo", specs: allSpecs },
             ];
 
-            const fileNames = "${{ steps.test-files.outputs.filenames }}"
-              .split(",")
-              .filter(Boolean)
-              .filter(testPath => !blacklist.some(blacklisted => testPath.includes(blacklisted)));
-
-            const fileCount = fileNames.length;
-
-            // subtract the special tests since they all get their own runners
-            const chunkSize = fileCount / (Number('${{ inputs.chunks }}') - specialTests.length);
+            const inputChunks = Number(process.env.chunks);
+            const regularChunks = inputChunks - specialTests.length;
 
+            const regularTests = new Array(regularChunks).fill(1)
+              .map((files, index) => ({
+                name: `e2e-group-${index + 1}`,
+              }));
 
-
-            const chunkedFiles = chunk(fileNames, chunkSize);
-
-            const regularTests = chunkedFiles.map((files, index) => ({
-              name: `e2e-group-${index + 1}`,
-              specs: files.join(","),
-              tags: '',
-            }));
-
+            // regular chunks need to come before special chunks
             const testSets = [...regularTests, ...specialTests];
 
             const config = testSets.map((options) => ({
               ...defaultOptions,
               ...options,
             }));
 
-            console.log("matrix config", config);
 
+            core.setOutput("regularChunks", JSON.stringify(regularChunks));
             core.setOutput("config", JSON.stringify({ include: config }));
```

Comment:
Nit (non-blocking): GitHub is complaining that using `setOutput` is deprecated:

![image](https://github.com/user-attachments/assets/28b6d2c0-31cf-4a50-b27a-1cca687b3edd)